File: src/main/java/de/hinundhergestellt/jhuh/sync/SyncProductRepository.java
package de.hinundhergestellt.jhuh.sync;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;
import java.util.UUID;
import java.util.stream.Stream;

@Repository
public interface SyncProductRepository extends JpaRepository<SyncProduct, UUID> {

    Stream<SyncProduct> findAllBy();

    Optional<SyncProduct> findByShopifyId(String shopifyId);
}

File: src/main/java/de/hinundhergestellt/jhuh/sync/SyncProductRepository.kt
package de.hinundhergestellt.jhuh.sync

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.UUID
import java.util.stream.Stream

@Repository
interface SyncProductRepository : JpaRepository<SyncProduct, UUID> {

    fun findAllBySyncedIsTrue(): Stream<SyncProduct>

    fun findByShopifyId(shopifyId: String): SyncProduct?

    fun findByArtooId(artooId: String): SyncProduct?

    fun findByVariantsBarcodeIn(barcode: List<String>): SyncProduct?
}
